### DISCUSSION TITLE
Need help defining two question to code in python for my studys

### DISCUSSION THREAD

**Discussion by KamiKamsn** [community/community#76364] [Programming Help]
### Body

Dear community,

this semester I started a course where we learn to code kind of AI in Python. For me its looks like more machine learning. But nvm.

So the task as final exam is to code a programm which can answer to questions about a data set:

_

‒ Two individual / unique research questions per student are required
 Procedure:
‒ Students search themselves for large and relevant data sets
‒ Students define two questions that should be answered for the selected data sets
‒ Lecturers check the data sets and the corresponding questions (such that problems are
difficult enough but not too difficult)
‒ Implementation of the solutions by the students on their own systems using the
presented libraries and methods of the lecture
- The resulting program code (as *.py) and the corresponding program execution need to
be analyzed regarding the run-time behavior. Which parameters influence the run-time in
which way?
‒ Code analysis and run-time behavior evaluation need to be executed per research
question.

The methods we learn and shall use:
- Data and data preparation (Pandas etc.) 
- Classification I (Support Vector Machines)
- Classification II (Decision trees, Random forests) 
- Clustering (kmeans, DBScan) 
- Testing and quality assurance (run time analysis) 
- Dimension reduction, anomaly detection
- Neural Networks 
- raining deep learning networks 
- Pipelines and MLOps 
- 
As im doing all the homework, I dont think that the coding part is my challenge.

**My problem is, to define two question which would fit the requirements.** Can you give me examples? The question should not be answerable by statistic.

For example I choose this dataset:
https://www.kaggle.com/datasets/nelgiriyewithana/billionaires-statistics-dataset/discussion
=> But I dont know which question could I define for this, which can be solved by the methods above.

I am also open for new datasets.

Thanks in advance!

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by johndoe-o** [comment]
> ### Body
> 
> Dear community,
> 
> this semester I started a course where we learn to code kind of AI in Python. For me its looks like more machine learning. But nvm.
> 
> So the task as final exam is to code a programm which can answer to questions about a data set:
> 
> _
> 
> ‒ Two individual / unique research questions per student are required
>  Procedure:
> ‒ Students search themselves for large and relevant data sets
> ‒ Students define two questions that should be answered for the selected data sets
> ‒ Lecturers check the data sets and the corresponding questions (such that problems are
> difficult enough but not too difficult)
> ‒ Implementation of the solutions by the students on their own systems using the
> presented libraries and methods of the lecture
> - The resulting program code (as *.py) and the corresponding program execution need to
> be analyzed regarding the run-time behavior. Which parameters influence the run-time in
> which way?
> ‒ Code analysis and run-time behavior evaluation need to be executed per research
> question.
> 
> The methods we learn and shall use:
> - Data and data preparation (Pandas etc.) 
> - Classification I (Support Vector Machines)
> - Classification II (Decision trees, Random forests) 
> - Clustering (kmeans, DBScan) 
> - Testing and quality assurance (run time analysis) 
> - Dimension reduction, anomaly detection
> - Neural Networks 
> - raining deep learning networks 
> - Pipelines and MLOps 
> - 
> As im doing all the homework, I dont think that the coding part is my challenge.
> 
> **My problem is, to define two question which would fit the requirements.** Can you give me examples? The question should not be answerable by statistic.
> 
> For example I choose this dataset:
> https://www.kaggle.com/datasets/nelgiriyewithana/billionaires-statistics-dataset/discussion
> => But I dont know which question could I define for this, which can be solved by the methods above.
> 
> I am also open for new datasets.
> 
> Thanks in advance!
> 
> 
> ### Guidelines
> 
> - [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

Beforeal.love
